docs/api-docs/analytics/bodl-overview.md

```diff
@@ -0,0 +1,423 @@
+This snippet uses the unauthenticated Storefront API to request the item objects from a completed order and concatenates them into a single array of physical items, digital items, and gift certificates.
+
+```handlebars title="Sample Pixel Code Start: Purchase Event" lineNumbers
+<script>
+  fetch('/api/storefront/order/{{checkout.order.id}}', {
+    credentials: 'same-origin'
+  })
+  .then(function(response) {
+    return response.json();
+  })
+  .then(function(orderJson) {
+    var orderQty = 0;
+    var lineItems = [];
+
+    for (i = 0; i < orderJson.lineItems.physicalItems.length; i++) {
+      var thisItem = orderJson.lineItems.physicalItems[i];
+      orderQty += thisItem.quantity;
+      lineItems.push({
+        "content_id": thisItem.productId,
+        "content_name": thisItem.name,
+        "currency": orderJson.currency.code,
+        "price": thisItem.salePrice,
+        "value": thisItem.extendedSalePrice,
+        "quantity": thisItem.quantity,
+        "content_type": "product_group"
+      });
+    }
+
+    for (i = 0; i < orderJson.lineItems.digitalItems.length; i++) {
+      var thisItem = orderJson.lineItems.digitalItems[i];
+      orderQty += thisItem.quantity;
+      lineItems.push({
+        "content_id": thisItem.productId,
+        "content_name": thisItem.name,
+        "currency": orderJson.currency.code,
+        "price": thisItem.salePrice,
+        "value": thisItem.extendedSalePrice,
+        "quantity": thisItem.quantity,
+        "content_type": "product_group"
+      });
+    }
+
+    for (i = 0; i < orderJson.lineItems.giftCertificates.length; i++) {
+      var thisItem = orderJson.lineItems.giftCertificates[i];
+      orderQty += thisItem.quantity;
+      lineItems.push({
+        "content_id": thisItem.type,
+        "content_name": thisItem.name,
+        "currency": orderJson.currency.code,
+        "price": thisItem.amount,
+        "value": thisItem.amount,
+        "quantity": thisItem.quantity,
+        "content_type": "product_group"
+      });
+    }
+
+    sampleAnalyticsProvider.instance('<%= property_id %>').track('Purchase', {
+        "contents": lineItems,
+        "value": orderJson.orderAmount,
+        "quantity": orderQty,
+        "currency": orderJson.currency.code
+    });
+  });
+</script>
+```
+
+### Registration
+
+This snippet tracks account creation.
+
+```handlebars title="Sample Pixel Code Start: Registration" lineNumbers
+<script>
+  if (window.location.pathname.indexOf('/login.php') === 0 && BODL.getQueryParamValue('action') === 'account_created') {
+    sampleAnalyticsProvider.instance('<%= property_id %>').track('Registration');
+  }
+</script>
+```
+
+### Search
+This snippet tracks when an end-user searches for products. Please take note that there is a built-in tracker for the category. However, we have commented out the tracker due to known distortions, which cause reporting of only the first category ID. Only reactivate if needed; use it at your own risk.
```

Comment:
@becomevocal , maybe it'd be better if we just cut out the code about the category tracker bit for now?

Reply:
I think what James was going for with this approach is the prevention of future partner support issues? Like, this must have been an issue we've run into before he received questions about.

Comfortable removing category for search though since it's less likely you'd utilize category in search reporting. We have category examples elsewhere in this doc.

Reply:
Cool. Yeah, I'd prefer to remove that kind of stuff and then we can add to once we have a clearer design solution for the schema and how to utilize something like that.